projects/acryptos/vault/index.js

```diff
@@ -0,0 +1,251 @@
+    if (tokenBalanceResult.success) {
+      const valueInToken = tokenBalanceResult.output;
+      const vaultAddress = tokenBalanceResult.input.target;
+      balances[vaultToUnderlyingToken[vaultAddress]] = BigNumber(balances[vaultToUnderlyingToken[vaultAddress]]).plus(
+        valueInToken
+      );
+    }
+  });
+  return balances;
+}
+
+async function tvlInSwapVault(timestamp, block) {
+  const balances = {};
+  let vaultInfos = {};
+  const lpToVault = {};
+
+  // Get vaults' underlying liquidity pairs' addresses
+  const underlyingVaultAddressResults = await sdk.bsc.abi.multiCall({
+    calls: _.map(swapVaultAddresses, (address) => ({
+      target: address,
+    })),
+    abi: vaultAbi["token"],
+    block: block,
+  });
+  vaultInfos = underlyingVaultAddressResults.output.reduce((acc, t) => {
+    if (t.success) {
+      acc[t.input.target] = { lpAddress: t.output };
+      lpToVault[t.output] = t.input.target;
+    }
+    return acc;
+  }, vaultInfos);
+  debug(vaultInfos);
+
+  // Get vaults' balance
+  const vaultBalanceResults = await sdk.bsc.abi.multiCall({
+    calls: _.map(swapVaultAddresses, (address) => ({
+      target: address,
+    })),
+    abi: vaultAbi["balance"],
+    block: block,
+  });
+  vaultInfos = vaultBalanceResults.output.reduce((acc, t) => {
+    acc[t.input.target].balance = t.success ? t.output : 0;
+    return acc;
+  }, vaultInfos);
+  debug(vaultInfos);
+
+  // Get liquidity pairs' underlying token addresses
+  const token0Results = await sdk.bsc.abi.multiCall({
+    calls: _.map(swapVaultAddresses, (address) => ({
+      target: vaultInfos[address].lpAddress,
+    })),
+    abi: lpAbi["token0"],
+    block: block,
+  });
+  vaultInfos = token0Results.output.reduce((acc, t) => {
+    acc[lpToVault[t.input.target]].token0 = t.output;
+    if (t.success && !balances[t.output]) {
+      balances[t.output] = 0;
+    }
+    return acc;
+  }, vaultInfos);
+  debug(vaultInfos);
+
+  const token1Results = await sdk.bsc.abi.multiCall({
+    calls: _.map(swapVaultAddresses, (address) => ({
+      target: vaultInfos[address].lpAddress,
+    })),
+    abi: lpAbi["token1"],
+    block: block,
+  });
+  vaultInfos = token1Results.output.reduce((acc, t) => {
+    acc[lpToVault[t.input.target]].token1 = t.output;
+    if (t.success && !balances[t.output]) {
+      balances[t.output] = 0;
+    }
+    return acc;
+  }, vaultInfos);
+  debug(vaultInfos);
+
+  // Get liquidity pairs' reserves
+  const lpReservesResults = await sdk.bsc.abi.multiCall({
+    calls: _.map(swapVaultAddresses, (address) => ({
+      target: vaultInfos[address].lpAddress,
+    })),
+    abi: lpAbi["getReserves"],
+    block: block,
+  });
+  vaultInfos = lpReservesResults.output.reduce((acc, t) => {
+    if (t.success) {
+      const { 0: reserve0, 1: reserve1 } = t.output;
+      acc[lpToVault[t.input.target]].reserve0 = reserve0;
+      acc[lpToVault[t.input.target]].reserve1 = reserve1;
+    }
+    return acc;
+  }, vaultInfos);
```

Comment:
for the above code we may use sdk.bsc.swap.getReserveBalance instead?

Reply:
```sdk.bsc.swap.getReserveBalance``` will aggregate token0 and token1 from different addresses together, which is not suitable for the calculation.